arangod/Pregel/Reports.cpp

```diff
@@ -0,0 +1,127 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2020 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Lars Maier
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Reports.h"
+
+using namespace arangodb::pregel;
+
+ReportBuilder::~ReportBuilder() {
+  // this can not throw because we have allocate memory
+  manager.append(Report{ss.str(), level, std::move(annotations)});
```

Comment:
I don't get why this can't throw. `ss` is an `std::stringstream`, and its `str()` method creates a copy:

From https://en.cppreference.com/w/cpp/io/basic_stringstream/str:

> 1) Returns a copy of the underlying string. Equivalent to return rdbuf()->str();.

Reply:
Damn. Good catch. :disappointed: